base/file_name_utils.hpp

```diff
@@ -40,7 +40,12 @@ template <typename... Args>
 std::string JoinPath(std::string const & folder, Args &&... args)
 {
   if (folder.empty())
-    return {};
+  {
+    if (sizeof...(args) == 0)
+      return {};
+    else
+      return impl::JoinPath(std::forward<Args>(args)...);
+  }
 
   return AddSlashIfNeeded(folder) + impl::JoinPath(std::forward<Args>(args)...);
 }
```

Comment:
Can we do this?
```cpp
template <typename... Args>
std::string JoinPath(Args &&... args)
{
// ...
}
```

Reply:
Do you mean something along this line:

```
template <typename... Args>
std::string JoinPath(Args &&... args)
{
  return (AddSlashIfNeeded(args)+...);
}
```
Unfortunately not without modifying the behavior of the function `AddSlashIfNeeded`. Currently `AddSlashIfNedded` returns `"/"` in case an empty string is entered. This case is also specifically tested in the test, thus I assumed this behavior to be desired.